### ISSUE TITLE
Issue with building website

### ISSUE THREAD

**Issue by WikiRik** [sequelize/website#216]
`yarn build` fails at this moment because of an issue with the generated files. Thing is that I can't reproduce this when running `yarn docs` on a GitHub Codespace on the sequelize repo. @ephys looking at the files that have been changed recently on main I also do not see what could be causing this, but if somebody can find out relatively easily where the error is it's you

```
Error:  Error: Prevent writing to file that only differs in casing or query string from already written file.
This will lead to a race-condition and corrupted files on case-insensitive file systems.
/home/runner/work/website/website/build/api/v7/functions/json.html
/home/runner/work/website/website/build/api/v7/functions/JSON.html
```

For an example of failing run see https://github.com/sequelize/website/runs/7161805298?check_suite_focus=true

**Comment by WikiRik** [comment]
What I could find what could possible cause this is the following;

https://github.com/sequelize/sequelize/blob/e480d6e83632306c995b236dbe103398b7ebc637/src/sequelize.js#L1515-L1528

https://github.com/sequelize/sequelize/blob/e480d6e83632306c995b236dbe103398b7ebc637/src/data-types.d.ts#L367-L370

**Comment by ephys** [comment]
It's interesting that this fails now. We haven't changed the name of these two functions. Did TypeDoc change how it names its files?

**Comment by WikiRik** [comment]
That could be a thing, 0.23 did have some breaking changes but the actual issue is with docusaurus. We might need to check the output of the docs generation in the CI before we build

**Comment by ephys** [comment]
IMO docusaurus is right here, we should not have URLs that are different depending on its casing. I'm hoping this is something that can be changed in TypeDoc :/

**Comment by WikiRik** [comment]
What I'm not so sure about is why we have both DataTypes.JSON and JSON files in the TypeDoc. That should be merged, so [json](https://github.com/sequelize/sequelize/blob/e480d6e83632306c995b236dbe103398b7ebc637/src/sequelize.js#L1515-L1528) is the only one there (and preferably we also rename that function)

**Comment by WikiRik** [comment]
This was fixed by reverting typedoc to 0.22. We'll wait until the DataTypes migration PR has been merged and then we'll try typedoc 0.23 again
